File: scala/problem13.scala
#!/usr/bin/env scala
!#

import scala.io.Source

import java.math.BigInteger

val lines = Source.fromFile("../data/problem13.txt").getLines.filter( !_.isEmpty )
val numbers = lines.map( new BigInteger( _ ) )
val sum = numbers.reduceLeft(_ add _)

println( sum.toString.take(10) )

File: ruby/problem13.rb
#!/usr/bin/env ruby

File.open File.join( File.dirname(__FILE__), '..', 'data', 'problem13.txt' ) do |f|
  @numbers = f.readlines.map &:to_i
end

summ = @numbers.inject :+

first_digits = summ.to_s[0..9]

puts first_digits
